apl/tabular-operators/redact-operator.mdx

```diff
@@ -0,0 +1,150 @@
+---
+title: redact
+description: 'This page explains how to use the redact operator in APL.'
+---
+
+The `redact` operator in APL replaces sensitive or unwanted data in string fields using regular expressions. You can use it to sanitize log data, obfuscate personal information, or anonymize text for auditing or analysis. The operator allows you to define one or multiple regular expressions to identify and replace matching patterns. You can customize the replacement token, generate hashes of redacted values, or retain structural elements while obfuscating specific segments of data.
+
+This operator is useful when you need to ensure data privacy or compliance with regulations such as GDPR or HIPAA. For example, you can redact credit card numbers, email addresses, or personally identifiable information from logs and datasets.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+In Splunk SPL, data sanitization is often achieved using custom regex-based transformations or eval functions. The `redact` operator in APL simplifies this process by directly applying regular expressions and offering options for replacement or hashing.
+
+<CodeGroup>
+```sql Splunk example
+| eval sanitized_field=replace(field, "regex_pattern", "*")
+```
+
+```kusto APL equivalent
+| redact 'regex_pattern' on field
+```
+</CodeGroup>
+
+</Accordion>
+<Accordion title="ANSI SQL users">
+
+ANSI SQL typically requires a combination of functions like `REPLACE` or `REGEXP_REPLACE` for data obfuscation. APL’s `redact` operator consolidates these capabilities into a single, flexible command.
+
+<CodeGroup>
+```sql SQL example
+SELECT REGEXP_REPLACE(field, 'regex_pattern', '*') AS sanitized_field FROM table;
+```
+
+```kusto APL equivalent
+| redact 'regex_pattern' on field
+```
+</CodeGroup>
+
+</Accordion>
+</AccordionGroup>
+
+## Usage
+
+### Syntax
+
+```kusto
+| redact [replaceToken="*"] [replaceHash=false] [redactGroups=false] <regex>, (<regex>) [on Field]
+```
+
+### Parameters
+
+| Parameter       | Type     | Description                                                                 |
+|-----------------|----------|-----------------------------------------------------------------------------|
+| `replaceToken`  | string   | The character with which to replace matches. The default is `*`.                     |
```

Comment:
There is a slight quirk here, in that if the replaceToken is a single character, we replace all the characters that match, like:
```
| extend foo = "foo bar baz"
| redact replaceToken="x" "foo" on foo 
```
 => `"xxx bar baz"` 

but if the replaceToken is more than one character then we replace the whole matching string with that token, for example
```
| extend foo = "foo bar baz"
| redact replaceToken="abc" "foo" on foo
```
=> `"abc bar baz"` 

which can be useful if you want to indicate that something was redacted, for example
```
| extend foo = "foo bar baz"
| redact replaceToken="<REDACTED>" "bar" on foo 
```
=> `"foo <REDACTED> baz"`